src/theory/uf/theory_uf.cpp

```diff
@@ -532,17 +590,25 @@ void TheoryUF::computeCareGraph() {
     unsigned functionTerms = d_functionsTerms.size();
     for (unsigned i = 0; i < functionTerms; ++ i) {
       TNode f1 = d_functionsTerms[i];
-      Node op = f1.getOperator();
+      Node op;
+      unsigned arg_start_index;
+      if( f1.getKind()==kind::HO_APPLY ){
+        op = d_equalityEngine.getRepresentative( f1[0] );
+        arg_start_index = 1;
+      }else{
+        op = f1.getOperator();
+        arg_start_index = 0;
+      }
```

Comment:
Seems a bit kludgey to special-case HO_APPLY like this - why not just change getOperator to understand HO_APPLY and have HO_APPLY work the same as APPLY_UF?

Reply:
Notice that fundamentally HO_APPLY needs to be handled differently than APPLY_UF since the "functions" they represent wrt the care graph occur in different places (APPLY_UF in the operator, and HO_APPLY as the first argument).  Also, we cannot treat HO_APPLY itself as a normal function since it is parametric.

I've made "get operator for apply term" and "get argument start index for apply term" into functions in my latest commit.